spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonTimeoutAutoConfiguration.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springboot.autoconfig;
+
+import org.axonframework.config.Configurer;
+import org.axonframework.config.ConfigurerModule;
+import org.axonframework.messaging.timeout.HandlerTimeoutHandlerEnhancerDefinition;
+import org.axonframework.messaging.timeout.TaskTimeoutSettings;
+import org.axonframework.messaging.timeout.UnitOfWorkTimeoutInterceptor;
+import org.axonframework.springboot.TimeoutProperties;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.annotation.Order;
+
+/**
+ * Configures the timeout settings for message handlers.
+ *
+ * @since 4.11
+ * @author Mitchell Herrijgers
+ */
+@AutoConfiguration
+@EnableConfigurationProperties(value = {
+        TimeoutProperties.class
+})
+@ConditionalOnProperty(prefix = "axon.timeout", name = "enabled", havingValue = "true", matchIfMissing = true)
+public class AxonTimeoutAutoConfiguration {
+    @Bean
+    public HandlerTimeoutHandlerEnhancerDefinition messageTimeoutHandlerEnhancerDefinition(
+            TimeoutProperties properties
+    ) {
+        return new HandlerTimeoutHandlerEnhancerDefinition(
+                properties.getHandler().toMessageHandlerTimeoutConfiguration()
+        );
+    }
+
+    @Bean
+    public ConfigurerModule axonTimeoutConfigurerModule(TimeoutProperties properties) {
+        return new AxonTimeoutConfigurerModule(properties);
+    }
+
+    @Order()
```

Comment:
Curious: why have the `@Order` annotation here?

Reply:
Leftover from experiment, before I discovered a `ConfigurerModule` could implement the order method